packages/react-dom/src/__tests__/findDOMNode-test.js

```diff
@@ -51,15 +53,23 @@ describe('findDOMNode', () => {
     }
 
     const container = document.createElement('div');
-
-    const myNodeA = ReactDOM.render(<MyNode />, container);
-    const a = ReactDOM.findDOMNode(myNodeA);
+    const root = ReactDOMClient.createRoot(container);
+
+    const nodeRef = React.createRef();
+    await act(async () => {
+      root.render(<MyNode ref={nodeRef} />);
+    });
+    const nodeAfterInitialRender = nodeRef.current;
+    const a = ReactDOM.findDOMNode(nodeAfterInitialRender);
     expect(a).toBe(null);
 
-    const myNodeB = ReactDOM.render(<MyNode flag={true} />, container);
-    expect(myNodeA === myNodeB).toBe(true);
+    await act(async () => {
+      root.render(<MyNode flag={true} ref={nodeRef} />);
+    });
+    const nodeAfterUpdate = nodeRef.current;
+    expect(nodeAfterInitialRender === nodeAfterUpdate).toBe(true);
```

Comment:
I'm not 100% sure that this is equivalent to the previous version of the test.

Previous version:
```
nodeA = ReactDOM.render(<Component>)
nodeB = ReactDOM.render(<UpdatedComponent>)
assert(nodeA === nodeB)
```
this version:
```
const ref = React.createRef()
root.render(<Something ref={ref} />)
const componentAfterInitialRender = ref.current
root.render(<UpdatedSomething ref={ref} />)
const componentAfterUpdate= ref.current
assert(componentAfterInitialRender === componentAfterUpdate)
```